include/ensmallen_bits/moead/moead_impl.hpp

```diff
@@ -283,8 +308,9 @@ Optimize(std::tuple<ArbitraryFunctionType...>& objectives,
     } // End of pass over all subproblems.
 
     //  The final population itself is the best front.
-    const std::vector<arma::uvec> frontIndices { arma::shuffle(
-        arma::linspace<arma::uvec>(0, populationSize - 1, populationSize)) };
+    // const std::vector<UVecType> frontIndices { shuffle(
+    //     linspace<UVecType>(0, populationSize - 1, populationSize)) };
+    const std::vector<UVecType> frontIndices;
```

Comment:
Is `frontIndices` only used by the `GenerationalStepTaken()` callback?  If so, and if the intention is just that that is the set of the population that is in the front, maybe it makes more sense to keep that on the CPU too.  But I am not deeply familiar with MOEAD.

Reply:
Fixed in [a9ce40c](https://github.com/mlpack/ensmallen/pull/425/commits/a9ce40c322030b65a9efffa9841ae8db53fd2cb7)